nemo/collections/nlp/modules/common/transformer/transformer_decoders.py

```diff
@@ -172,6 +180,11 @@ def forward(
             memory_states = self._get_memory_states(decoder_states, decoder_mems_list, i + 1)
             cached_mems_list.append(memory_states)
 
+        if self.pre_ln and self.pre_ln_final_layer_norm:
```

Comment:
You may check it like this:
```
if self.final_layer_norm is not None:
            decoder_states = self.final_layer_norm(decoder_states)
            memory_states = self._get_memory_states(decoder_states, encoder_mems_list, i + 1)
            cached_mems_list.append(memory_states)
```

Reply:
Yes I was overthinking maybe we gonna add more norm methods. yeah let me simply it